cmd/triggerRun/cmd/root_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"encoding/json"
+	"net/http/httputil"
+	"regexp"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestReadTrigger(t *testing.T) {
+	tri, err := readTrigger("../testdata/trigger.yaml")
+	if err != nil {
+		t.Fatal("", err)
+	}
+
+	bytes, err := json.Marshal(tri)
```

Comment:
actually instead of marshalling the `tri` to a JSON string and then comparing below, what we can do to simplify is just compare the structs.
readTrigger returns `[]*v1alpha1.Trigger` so the `expect` variable below should also be of the same type:
```
expect := []*v1alpha1.Trigger{{
  TypeMeta: metav1.TypeMeta{
     APIVersion: ....,
    Kind: "",
  },
 ObjectMeta: metav1.ObjectMeta{
    Name: "..."
 },
 Spec: ....
}}
```

Similar to https://github.com/tektoncd/triggers/blob/master/pkg/sink/sink_test.go#L227